core/src/test/java/org/infinispan/persistence/CacheLoaderFunctionalTest.java

```diff
@@ -86,6 +86,7 @@ protected ConfigurationBuilder getConfiguration() {
       ConfigurationBuilder cfg = new ConfigurationBuilder();
       cfg.persistence()
          .addStore(DummyInMemoryStoreConfigurationBuilder.class)
+            .segmented(true)
```

Comment:
Do we not want this test to run with segmented true and false?

Reply:
We should